tests/balloonpanel/balloonpanelview.js

```diff
@@ -0,0 +1,234 @@
+
+	describe( 'DOM bindings', () => {
+		describe( 'arrow', () => {
+			it( 'should reacts on view.model#arrow', () => {
+				expect( view.element.classList.contains( 'ck-balloon-panel_arrow_se' ) ).to.true;
+
+				view.model.set( 'arrow', 'sw' );
+
+				expect( view.element.classList.contains( 'ck-balloon-panel_arrow_sw' ) ).to.true;
+			} );
+		} );
+
+		describe( 'isVisible', () => {
+			it( 'should reacts on view.model#isvisible', () => {
+				expect( view.element.classList.contains( 'ck-balloon-panel_visible' ) ).to.false;
+
+				view.model.isVisible = true;
+
+				expect( view.element.classList.contains( 'ck-balloon-panel_visible' ) ).to.true;
+			} );
+		} );
+
+		describe( 'styles', () => {
+			it( 'should reacts on view.model#top', () => {
+				expect( view.element.style.top ).to.equal( '0px' );
+
+				view.model.top = 10;
+
+				expect( view.element.style.top ).to.equal( '10px' );
+			} );
+
+			it( 'should reacts on view.model#left', () => {
+				expect( view.element.style.left ).to.equal( '0px' );
+
+				view.model.left = 10;
+
+				expect( view.element.style.left ).to.equal( '10px' );
+			} );
+
+			it( 'should reacts on view.model#maxWidth', () => {
+				expect( view.element.style.maxWidth ).to.equal( '200px' );
+
+				view.model.maxWidth = 10;
+
+				expect( view.element.style.maxWidth ).to.equal( '10px' );
+			} );
+		} );
+	} );
+
+	describe( 'isVisible', () => {
+		it( 'should return view.model#isVisible value', () => {
+			expect( view.isVisible ).to.false;
+
+			view.model.isVisible = true;
+
+			expect( view.isVisible ).to.true;
+		} );
+	} );
+
+	describe( 'show', () => {
+		it( 'should set view.model#isVisible as true', () => {
+			view.show();
+
+			expect( view.isVisible ).to.true;
+		} );
+	} );
+
+	describe( 'hide', () => {
+		it( 'should set view.model#isVisible as false', () => {
+			view.model.isVisible = true;
+
+			view.hide();
+
+			expect( view.isVisible ).to.false;
+		} );
+
+		it( 'should set view.model#isVisible as false', () => {
+			it( 'should fires view.model#hide event', () => {
+				const hideSpy = sinon.spy();
+
+				view.model.on( 'hide', hideSpy );
+
+				view.hide();
+
+				expect( hideSpy.calledOnce ).to.true;
+			} );
+		} );
+	} );
+
+	describe( 'attachTo', () => {
+		const targetEl = document.createElement( 'div' );
+		targetEl.style.cssText = 'position: absolute; width: 50px; height: 50px;';
+
+		const limiter = { left: 0, top: 0, right: 300, bottom: 300 };
+
+		before( () => {
+			document.body.appendChild( targetEl );
+
+			document.body.style.minHeight = '350px';
+			document.body.style.minWidth = '350px';
+		} );
+
+		after( () => {
+			document.body.removeChild( targetEl );
+
+			document.body.style.minHeight = null;
+			document.body.style.minWidth = null;
+		} );
+
+		beforeEach( () => {
+			// Add some appearance to balloon panel element and append to document.
+			view.element.style.width = '150px';
+			view.element.style.height = '100px';
+			view.element.style.position = 'absolute';
```

Comment:
AFAIR `.ck-ballon-panel` has `position: absolute;` by default from the stylesheet.